xbmc/windowing/gbm/WinSystemGbm.cpp

```diff
@@ -346,7 +360,19 @@ bool CWinSystemGbm::SetHDR(const VideoPicture* videoPicture)
     return true;
   }
 
-  auto connector = drm->GetConnector();
+  if (connector->SupportsProperty("Colorspace"))
+  {
+    auto [result, value] =
+        connector->GetPropertyValue("Colorspace", DRMPRIME::GetColorimetry(*videoPicture));
+    if (result)
+    {
+      CLog::Log(LOGDEBUG, "CWinSystemGbm::{} - setting connector colorspace to {}", __FUNCTION__,
+                DRMPRIME::GetColorimetry(*videoPicture));
+      drm->AddProperty(connector, "Colorspace", value);
+      drm->SetActive(true);
```

Comment:
Keep in mind, that you are setting connector property here and not a plane or crtc property: At this point you do not know, if the connector (there are a lot more connectors than hdmi, btw) actually supports the colorspace you are setting here and might result in wrong colors.

Reply:
I'm not sure what you mean, we query the supported values and only set them if they are available.

Reply:
Really we should check the EDID but that can added after #19141

Reply:
> Really we should check the EDID but that can added after https://github.com/xbmc/xbmc/pull/19141

Yes, that certainly needs to be done - since https://github.com/torvalds/linux/blob/master/drivers/gpu/drm/drm_connector.c#L1905-L1922, just gives adds all possible colorspaces and there is nothing es done in https://github.com/torvalds/linux/blob/master/drivers/gpu/drm/i915/display/intel_connector.c#L283-L288 and with a screen which definitly doesn't support BT2020 `modetest -c `gives
```
	65 Colorspace:
		flags: enum
		enums: Default=0 SMPTE_170M_YCC=1 BT709_YCC=2 XVYCC_601=3 XVYCC_709=4 SYCC_601=5 opYCC_601=6 opRGB=7 BT2020_CYCC=8 BT2020_RGB=9 BT2020_YCC=10 DCI-P3_RGB_D65=11 DCI-P3_RGB_Theater=12
		value: 0
```